Sources/OktaIdxAuth/Internal/Implementation.swift

```diff
@@ -219,6 +227,45 @@ extension OktaIdxAuth.Implementation: OktaIdxAuthImplementation {
         }
     }
     
+    @available(iOSApplicationExtension 13.0, *)
+    @objc func socialAuth(with options: OktaIdxAuth.SocialAuth.Options, completion: OktaIdxAuth.ResponseResult<OktaIdxAuth.Response>?) {
```

Comment:
I suggest extracting these functions into a common private method that can handle these cases.

My suggestion is that the private method can have `options` be a nullable value; when null, you fall back to the iOS 12 implementation, otherwise you carry on with the iOS 13 one.

Reply:
I would like to merge into one method. But the reasons is availability.

`SocialOptions` has the property `let presentationContext: ASWebAuthenticationPresentationContextProviding`. 
It's available only staring from iOS 13. 
Swift forbids making availability for stored properties. That's why `available` is written for `SocialOptions`. 

<img width="1390" alt="Screenshot 2021-04-29 at 12 33 43" src="https://user-images.githubusercontent.com/77254185/116531293-b1460580-a8e7-11eb-9f53-04483415096f.png">

Because of that, we cannot make `socialAuth(with options: OktaIdxAuth.SocialAuth.Options` available for iOS 12. And we should have two different methods.